fix: Markdown paragraph spacing issue in recipe notes

cnorteye opened the pull request:
<!--
  This template provides some ideas of things to include in your PR description.

  To start, try providing a short summary of your changes in the Title above. We follow Conventional Commits syntax, please ensure your title is prefixed with one of:
  - `feat: `
  - `fix: `
  - `docs: `
  - `chore: `
  - `dev:`

  If a section of the PR template does not apply to this PR, then delete that section.

  PLEASE READ:
  -------------------------
  Mealie is moving to a regular, automatic release schedule. This means that all PRs should be in a
  stable state, ready for release. This includes:

  - Ensuring new tests have been added to cover new features, or to prevent regressions.
  - Work is fully complete and usable

 -->

## What type of PR is this?
- bug


## What this PR does / why we need it:
This PR addresses an issue where single-line breaks in recipe notes caused incorrect paragraph spacing. The following changes were made:
- Added a `preprocessMarkdown` function to `RecipeNotes.vue` to convert single line breaks into double line breaks for proper Markdown rendering.
- Updated `SafeMarkdown` usage in `RecipeNotes.vue` to preprocess text before rendering.
These changes ensure that paragraphs are spaced correctly when rendering recipe notes.

Before: 
![Screenshot 2024-12-02 at 10 58 14 PM](https://github.com/user-attachments/assets/4525fca5-182a-47c7-aed2-a9c41c132826)
![Screenshot 2024-12-02 at 10 58 28 PM](https://github.com/user-attachments/assets/e96000f2-2926-455b-b2fc-2783c9748b0e)


After: 
![Screenshot 2024-12-02 at 10 51 27 PM](https://github.com/user-attachments/assets/6a992372-12f7-45f0-a7e1-e62ee16f1710)
![Screenshot 2024-12-02 at 10 51 41 PM](https://github.com/user-attachments/assets/61a154af-7f09-40db-9e01-1da66f4031c9)


## Which issue(s) this PR fixes:
Fixes #2854
Fixes #2846

## Testing
 Since this was a UI improvement, I just observed the changes

michael-genson commented:
Hey there! Changes look good in the screenshots, but I think something's wrong with the branch/PR. I'm only seeing changes to the poetry lock file, rather than the frontend changes. Can you double check the branch?

cnorteye commented:
> Hey there! Changes look good in the screenshots, but I think something's wrong with the branch/PR. I'm only seeing changes to the poetry lock file, rather than the frontend changes. Can you double check the branch?

Yes, I look into that. Thank you

michael-genson commented:
It looks like you changed a bunch of the dev container config and added NPM. It doesn't look like this PR needs any dependency changes, so please remove changes to those files from this PR. If I'm reading this correctly, the only changes should be those made in `frontend/components/global/SafeMarkdown.vue`.

It may be easier to start fresh with a new branch off of `mealie-next`, rather than removing changes from each file

cnorteye commented:
> It looks like you changed a bunch of the dev container config and added NPM. It doesn't look like this PR needs any dependency changes, so please remove changes to those files from this PR. If I'm reading this correctly, the only changes should be those made in `frontend/components/global/SafeMarkdown.vue`.
> 
> It may be easier to start fresh with a new branch off of `mealie-next`, rather than removing changes from each file

Let me know if I did it correctly. I moved all the changes I made and only updated the `frontend/components/global/SafeMarkdown.vue`. When trying to commit, my changes kept failing the fix end of file pre-commit hooks so I had to make some minor changes to the formatting within the `SafeMarkdown.vue` file.

michael-genson commented:
I'm still seeing changes to 11 files: https://github.com/mealie-recipes/mealie/pull/4664/files

cnorteye commented:
Yes, I am not sure how to remove those files. I tried your recommendation of working off another branch, but when I committed my changes it still shows the 10 files from the prior commit. I searched to see if there was a way to undo a commit and none of the answers were helpful. It seems like maybe closing this PR and reopening another might be better? Please let me know.

cnorteye commented:
@michael-genson I opened a draft pull request #4687. I had accidentally closed this PR attempting to update the files. Let me know upon your review if the other PR is good. Thanks for your patience while I tried to figure this out :)

michael-genson commented:
No worries! That one looks much better :) I should have time to review early next week (out of town this weekend). Feel free to take it out of draft when it's ready!

cnorteye commented:
Sounds good! I have marked it for review.
